alpine/hotspot/Dockerfile

```
@@ -16,19 +16,30 @@ RUN apk add --no-cache \
     && rm -fr /var/cache/apk/* \
     && /usr/bin/jdk-download.sh alpine
 
-ENV PATH=/opt/jdk-${JAVA_VERSION}/bin:$PATH
+ENV PATH="/opt/jdk-${JAVA_VERSION}/bin:${PATH}"
 
 # Generate smaller java runtime without unneeded files
 # for now we include the full module path to maintain compatibility
 # while still saving space (approx 200mb from the full distribution)
-# Arguments to jlink are specific to Java 17 jlink
-RUN jlink \
-        --add-modules ALL-MODULE-PATH \
-        --strip-java-debug-attributes \
-        --no-man-pages \
-        --no-header-files \
-        --compress=2 \
-        --output /javaruntime
+RUN if test "${TARGETPLATFORM}" != 'linux/arm/v7'; then \
+    case "$(jlink --version 2>&1)" in \
+      # jlink version 11 has less features than JDK17+
+      # the compression argument is different for JDK21
+      "11."*) set -- "--strip-debug" "--compress=2" ;; \
+      "21."*) set -- "--strip-java-debug-attributes" "--compress=zip-6" ;; \
+      *) set -- "--strip-java-debug-attributes" "--compress=2" ;; \
```

```suggestion
      # jlink version 11 has less features than JDK17+
      "11."*) set -- "--strip-debug" "--compress=2" ;; \
      "17."*) set -- "--strip-java-debug-attributes" "--compress=2" ;; \
      # the compression argument is different for JDK21
      "21."*) set -- "--strip-java-debug-attributes" "--compress=zip-6" ;; \
      *) echo "ERROR: unmanaged jlink version pattern && exit 1 ;; \
```
Suggestion to make this switch a bit more explicit, WDYT?

---

Addressed in https://github.com/jenkinsci/docker/pull/1857/commits/f01ba66e5884a24afa738820aeda629eed6c5bf0

Also added https://github.com/jenkinsci/docker/pull/1857/commits/d36dc7054951809a15ac1f74cd73e9f48449ced9 for Windows to be iso with Linux.